source/common/upstream/load_balancer_impl.cc

```diff
@@ -1301,7 +1301,8 @@ HostConstSharedPtr LeastRequestLoadBalancer::unweightedHostPick(const HostVector
 
   // Do full scan if it's required explicitly or the number of choices is equal to or larger than
   // the hosts size.
-  if ((hosts_to_use.size() <= choice_count_) || enable_full_scan_) {
+  if ((full_scan_if_host_num_less_than_choice_num_ && hosts_to_use.size() <= choice_count_) ||
```

Comment:
Is there a reported issue for this, and what is the reason? Thanks!

Reply:
Issue for public reference: Istio disables cluster stats so it causes this PR changes load balancing to always pick the first host. Due to the DNS traffic concentration, there's some unexpected DNS failure as a result. We were not able to get Istio CI to deploy it https://github.com/istio/istio/pull/47694.

Reply:
I think we can revert that PR temporarily until we figure out a new API for that. (Today is the last day that could do a break change to the API)
cc @kyessenov

Reply:
Yeap, let's do that, and then open it again with a safe default (e.g. by putting && instead of || when the flag is enabled).